apps/cowswap-frontend/src/legacy/components/Header/Polling.tsx

```diff
@@ -157,25 +154,18 @@ const Wrapper = styled.div`
   }
 `
 
-const DEFAULT_MS_BEFORE_WARNING = ms`10m`
-const NETWORK_HEALTH_CHECK_MS = ms`10s`
-
 export function Polling() {
   const { chainId } = useWalletInfo()
   const blockNumber = useBlockNumber()
   const [isMounting, setIsMounting] = useState(false)
   const [isHover, setIsHover] = useState(false)
-  const machineTime = useMachineTimeMs(NETWORK_HEALTH_CHECK_MS)
-  const blockTime = useCurrentBlockTimestamp()
+  const isOnline = useIsOnline()
   const theme = useTheme()
 
   const ethGasPrice = useGasPrice()
   const priceGwei = ethGasPrice ? JSBI.divide(ethGasPrice, JSBI.BigInt(1000000000)) : undefined
 
-  const waitMsBeforeWarning =
-    (chainId ? getChainInfo(chainId)?.blockWaitMsBeforeWarning : DEFAULT_MS_BEFORE_WARNING) ?? DEFAULT_MS_BEFORE_WARNING
-
-  const warning = Boolean(!!blockTime && machineTime - blockTime.mul(1000).toNumber() > waitMsBeforeWarning)
+  const warning = !isOnline
```

Comment:
all these changes look unrelated. What is this logic that was removed?

Reply:
Great question.
`useCurrentBlockTimestamp` depends on uniswap multicall, so I decided to fully replace it by `useIsOnline`.
Essentially, it should be equivalent and even better.
Before the component was checking time gap between last block timestamp. Now we directly use browser API to detect if the browser is online